### PR TITLE
feat: add subscriptions

### PR DESCRIPTION
**Pull request by sdanialraza**
**Please describe the changes this PR makes and why it should be merged:**
Adds subscriptions for core

Depends on changes from:
- https://github.com/discord/discord-api-docs/pull/7109
- https://github.com/discord/discord-api-docs/pull/7110
- https://github.com/discordjs/discord-api-types/pull/1078
- https://github.com/discordjs/discord-api-types/pull/1105

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/core/src/api/monetization.ts [hunk 1]
```diff
@@ -17,17 +20,53 @@ export class MonetizationAPI {
...
-	 * @see {@link https://discord.com/developers/docs/monetization/skus#list-skus}
+	 * @see {@link https://discord.com/developers/docs/resources/sku#list-skus}
 	 * @param options - The options for fetching the SKUs.
 	 */
 	public async getSKUs(applicationId: Snowflake, { signal }: Pick<RequestData, 'signal'> = {}) {
 		return this.rest.get(Routes.skus(applicationId), { signal }) as Promise<RESTGetAPISKUsResult>;
 	}
 
+	/**
+	 * Fetches the subscriptions for a SKU.
```
Comment (Jiralite) [inline packages/core/src/api/monetization.ts:31]:
```suggestion
	 * Fetches the subscriptions for an SKU.
```

File: packages/core/src/api/monetization.ts [hunk 2]
```diff
@@ -17,17 +20,53 @@ export class MonetizationAPI {
...
+		{ signal }: Pick<RequestData, 'signal'> = {},
+	) {
+		return this.rest.get(Routes.skuSubscriptions(skuId), {
+			signal,
+			query: makeURLSearchParams(query),
+		}) as Promise<RESTGetAPISKUSubscriptionsResult>;
+	}
+
+	/**
+	 * Fetches the subscription for a SKU.
```
Comment (Jiralite) [inline packages/core/src/api/monetization.ts:50]:
```suggestion
	 * Fetches the subscription for an SKU.
```

**Review by Jiralite — DISMISSED** [review]

**Review by almeidx — CHANGES_REQUESTED** [review]
The new subscription events need to be added here

https://github.com/sdanialraza/discord.js/blob/f5761a9593ff6c691d0c752db2f97fd25dc46f7d/packages/core/src/client.ts#L96

File: packages/core/src/api/monetization.ts [hunk 3]
```diff
@@ -17,17 +20,53 @@ export class MonetizationAPI {
...
+	/**
+	 * Fetches the subscription for an SKU.
+	 *
+	 * @see {@link https://discord.com/developers/docs/resources/subscription#get-sku-subscription}
+	 * @param skuId - The SKU id to fetch subscription for
+	 * @param options - The options for fetching the subscription
+	 */
+	public async getSKUSubscription(
+		skuId: Snowflake,
+		subscriptionId: Snowflake,
```
Comment (almeidx) [inline packages/core/src/api/monetization.ts:58]: Missing in the TSDoc

**Review by almeidx — CHANGES_REQUESTED** [review]

File: packages/core/src/api/monetization.ts [hunk 4]
```diff
@@ -17,17 +20,55 @@ export class MonetizationAPI {
...
+		{ signal }: Pick<RequestData, 'signal'> = {},
+	) {
+		return this.rest.get(Routes.skuSubscriptions(skuId), {
+			signal,
+			query: makeURLSearchParams(query),
+		}) as Promise<RESTGetAPISKUSubscriptionsResult>;
+	}
+
+	/**
+	 * Fetches the subscription for an SKU.
```
Comment (almeidx) [inline packages/core/src/api/monetization.ts:51]:
```suggestion
	 * Fetches a subscription for an SKU.
```

File: packages/core/src/api/monetization.ts [hunk 5]
```diff
@@ -17,17 +20,55 @@ export class MonetizationAPI {
...
+	 * @see {@link https://discord.com/developers/docs/resources/sku#list-skus}
+	 * @param applicationId - The application id to fetch SKUs for
 	 * @param options - The options for fetching the SKUs.
 	 */
 	public async getSKUs(applicationId: Snowflake, { signal }: Pick<RequestData, 'signal'> = {}) {
 		return this.rest.get(Routes.skus(applicationId), { signal }) as Promise<RESTGetAPISKUsResult>;
 	}
 
+	/**
+	 * Fetches the subscriptions for an SKU.
```
Comment (almeidx) [inline packages/core/src/api/monetization.ts:32]:
```suggestion
	 * Fetches subscriptions for an SKU.
```

**Review by almeidx — APPROVED** [review]

**Review by Jiralite — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**